stubs/aiofiles/aiofiles/os.pyi

```diff
@@ -1,21 +1,71 @@
 import sys
 from _typeshed import StrOrBytesPath
+from asyncio.events import AbstractEventLoop
+from concurrent.futures._base import Executor
 from os import stat_result
 from typing import Sequence, Union, overload
 
 _FdOrAnyPath = Union[int, StrOrBytesPath]
 
-async def stat(path: _FdOrAnyPath, *, dir_fd: int | None = ..., follow_symlinks: bool = ...) -> stat_result: ...
+async def stat(
+    path: _FdOrAnyPath,
+    *,
+    dir_fd: int | None = ...,
+    follow_symlinks: bool = ...,
+    loop: AbstractEventLoop | None = ...,
+    executor: Executor | None = ...,
```

Comment:
chose this typing as the docs suggest that this is either `ProcessPoolExecutor` or `ThreadPoolExecutor`: https://github.com/python/cpython/blob/main/Doc/library/asyncio-eventloop.rst

```
## Options:

          # 1. Run in the default loop's executor:
          result = await loop.run_in_executor(
              None, blocking_io)
          print('default thread pool', result)

          # 2. Run in a custom thread pool:
          with concurrent.futures.ThreadPoolExecutor() as pool:
              result = await loop.run_in_executor(
                  pool, blocking_io)
              print('custom thread pool', result)

          # 3. Run in a custom process pool:
          with concurrent.futures.ProcessPoolExecutor() as pool:
              result = await loop.run_in_executor(
                  pool, cpu_bound)
              print('custom process pool', result)
```

Reply:
N.B. most of asyncio stubs type executor as `Any` so happy to either:

a) Change the executor typing here to `Any` as well. Maybe the motivation is that people might implement their own executors that don't inherit from `Executor`?
b) Use the `Executor` typing here, since that's what the docs imply, and then we can refine the executor typing in asyncio in a subsequent PR?

Reply:
This should be consistent with stdlib. If we change asyncio later, in a separate PR, we can update aiofiles too.

Reply:
will do!